cds-snc/notification-admin
# Better error messaging when blocked by Trial

# Summary | Résumé

[TRELLO](https://trello.com/c/U7h3nIX5/371-level-1-update-trial-to-live-messaging-notify-wide): Improves the error messages upon sending a message.

# Test instructions | Instructions pour tester la modification

Manual tests: 

1. Get around the screens to send messages.
2. Verify the messages are correctly changed. 

# Screenshots

## Trying to send an email to someone outside of the service team

![image](https://user-images.githubusercontent.com/805567/112511018-64ab5f80-8d68-11eb-809b-6dea427728bb.png)

## Clicking the _request to go live_ link with permissions to manage the service

![image](https://user-images.githubusercontent.com/805567/112511375-bbb13480-8d68-11eb-8b4d-4381f08456fc.png)

## Clicking the _request to go live_ link without the permissions to manage the service

![image](https://user-images.githubusercontent.com/805567/112515926-1c427080-8d6d-11eb-915f-9bb3e764523e.png)

Notice I didn't include the service manager's email per the design. I want to do this in the Trial banner work in an incremental fashion.

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their review | Voici une suggestion de liste de vérification comprenant des questions que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce que ça devrait être divisé en de plus petites demandes de tirage (« pull requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce changement (fichier README, etc.)?
